### PR TITLE
Handle subroadmap

### LINKED ISSUE #3: Extract content from roadmap pages
- Roadmap
```
{
        "id": "npnMwSDEK2aLGgnuZZ4dO",
        "type": "subtopic",
        "position": {"x": 2039.941514522558, "y": 229.3007298193524},
        "selected": False,
        "data": {
            "label": "Go",
            "style": {
                "fontSize": 17,
                "justifyContent": "flex-start",
                "textAlign": "center",
            },
            "oldId": "QCdemtWa2mE78poNXeqzr",
            "legend": {
                "id": "RkcQf9vafpvjAVpo02XhQ",
                "color": "#2b78e4",
                "label": "Personal Recommendation / Opinion",
                "position": "left-center",
            },
        },
        "zIndex": 999,
        "width": 107,
        "height": 49,
        "positionAbsolute": {"x": 2039.941514522558, "y": 229.3007298193524},
        "dragging": False,
        "style": {"width": 107, "height": 49},
        "resizing": False,
        "selectable": True,
        "focusable": True,
}
```

URL: https://roadmap.sh/devops/go@npnMwSDEK2aLGgnuZZ4dO
Structure: https://roadmap.sh/devops/data.label@id

- Subroadmap
```
{
        "ID": "917",
        "typeID": "__group__",
        "zOrder": "36",
        "measuredW": "222",
        "measuredH": "42",
        "w": "222",
        "h": "42",
        "x": "359",
        "y": "422",
        "properties": {"controlName": "100-python-basics:basic-syntax"},
}
```
topic and subtopic come from properties.controlName
URL: https://roadmap.sh/python/python-basics/typecasting-exceptions
Structure: https://roadmap.sh/python/topic/subtopic

### PR DESCRIPTION
**Pull request by npujol**
- [x] Handle sub roadmaps

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -166,3 +166,6 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+# Exclude nix build
+result/*
```
Comment (leiserfg) [inline .gitignore]: `result` is enough

File: app/main.py [hunk 2]
```diff
@@ -6,21 +6,28 @@
...
+
+from app.serializers import Roadmap, SubRoadmap
 
 logging.basicConfig(
-    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
 )
 
-logger: logging.Logger = logging.getLogger(name=__name__)
+logger: logging.Logger = logging.getLogger(name=__name__.split(sep=".")[-1])
```
Comment (leiserfg) [inline app/main.py]: This makes it hard to filter loggers

File: app/serializers.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                if node.properties is None:
+                    continue
+                control_name: str | None = node.properties.controlName
+                # Exclude reference to other roadmaps
+                if control_name is None or "roadmap.sh" in control_name:
+                    continue
+
+                control_name_parts: list[str] = control_name.split(sep=":")
+                # Topic example: 100-python-basics
+                if len(control_name_parts) == 1:
```
Comment (leiserfg) [inline app/serializers.py]: I think these two branches do almost the same thing 🤔

**Review by leiserfg — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (npujol) [reply .gitignore]: Done

File: app/main.py (hunk 2 above)
Comment (npujol) [reply app/main.py]: Done

**Review by leiserfg — APPROVED** [review]

**Pull request merged**